tools/test/audio/comp_run.sh

```diff
@@ -28,6 +28,7 @@ FN_IN=input.raw
 FN_OUT=output.raw
 FN_TRACE:=trace.txt # This is default value if FN_TRACE is not set via -e option
 VALGRIND=yes
+XTRUN=
```

Comment:
Why not just test for XTENSA_TOOLS_ROOT like all other scripts?

Reply:
I'm running both x86 native (a lot faster) and selected tests with xt-run binary. I'd not like to set/unset XTENSA_TOOLS_ROOT to select what's launched. At Octave command level xt-run or native is selected this way:

`process_test('eq-iir', 32, 32, 48000, 1, 'xt-run --turbo');`
or

`process_test('eq-iir', 32, 32, 48000, 1)`

The --turbo is a faster and less accurate simulation. There's a lot more options possible for xt-run so I'd like to pass the 6th arg directly for xt-run launch.

But I'd like to add to comp_run.sh the xt-environment discovery with XTENSA_TOOLS_ROOT etc. what's needed for Zephyr build. Now it's depending on all XTENSA environment variables including binaries PATH set up. It comes from my old xcc xtos build env setup and obviously other developers don't  have a similar setup.

Reply:
> I'm running both x86 native (a lot faster) and selected tests with xt-run binary. I'd not like to set/unset XTENSA_TOOLS_ROOT to select what's launched.

Fair enough; the switch between Zephyr SDK and XT-XCC is not the same as the switch between native x86 and XT-XCC.

> But I'd like to add to comp_run.sh the xt-environment discovery with XTENSA_TOOLS_ROOT etc. what's needed for Zephyr build. Now it's depending on all XTENSA environment variables including binaries PATH set up. It comes from my old xcc xtos build env setup and obviously other developers don't have a similar setup.

+1, it should be extracted from `scripts/xtensa-build-all.sh` into a new `source scripts/xtensa_env.sh` file.

Reply:
I solved this by adding export of file xtrun_env.sh from scripts/rebuild-testbench.sh. The comp_run.sh sources the file and therefore does not need any environment variables or paths configuration. Also the xt-run core etc. matches then always the build. I'll push next the version with this update.